File: tests/xcalablemp/not-implemented-yet/global-view/template/F/template_fix3.f90
      program main
      interface
         subroutine sub(x, N)
!$xmp nodes p(*)
!$xmp template t(:)
!$xmp distribute t(cyclic(3)) onto p
         integer N
         integer,allocatable:: x(:)
!$xmp align x(i) with t(i)
         end subroutine
      end interface
        
      include 'xmp_lib.h'
!$xmp nodes p(*)
!$xmp template t(:)
!$xmp distribute t(cyclic(3)) onto p
      integer N, s
      integer,allocatable:: a(:)
!$xmp align a(i) with t(i)
      character(len=2) result

      N = 0
      do i=1, 1000
         N = N+1
      enddo
      
!$xmp template_fix (cyclic(3)) t(N)
      allocate(a(N))

      call sub(a,N)
         
      s = 0
!$xmp loop (i) on t(i) reduction(+: s)
      do i=1, N
         s = s+a(i)
      enddo

      result = 'OK'
      if(s .ne. 500500) then
         result = 'NG'
         print *, s
      endif

      print *, xmp_all_node_num(), 'testp003.f ', result
      deallocate(a)

      end

      subroutine sub(x, N)
!$xmp nodes p(*)
!$xmp template t(:)
!$xmp distribute t(cyclic(3)) onto p
      integer N
      integer,allocatable:: x(:)
!$xmp align x(i) with t(i)

!$xmp loop (i) on t(i)
      do i=1, N
         x(i) = i
      enddo

      end

File: tests/xcalablemp/NOT-IMPLEMENTED/294/template_fix3.c
int sub(int *x,int N){
#pragma xmp nodes p(*)
#pragma xmp template t(0:N-1)
#pragma xmp distribute t(cyclic(3)) onto p
#pragma xmp align x[i] with t(i)
   return 0;
}

#include<xmp.h>
#include<stdio.h>
#include<stdlib.h>
#pragma xmp nodes p(*)
#pragma xmp template t(:)
#pragma xmp distribute t(cyclic(3)) onto p
int N,s,*a,i,*x,result=0;
#pragma xmp align a[i] with t(i)

int main(void)
{
  N = 1000;
#pragma xmp template_fix(cyclic(3)) t(0:N-1)
  a = (int *)malloc(N);
  sub(a,N);   

#pragma xmp loop (i) on t(i)
  for(i=0;i<N;i++)
    a[i] = i;

   s = 0;
#pragma xmp loop (i) on t(i) reduction(+:s)
   for(i=0;i<N;i++)
      s = s+a[i];

   if(s != 499500)
     result = -1;

   return 0;
}
               
      
   
